arc/analysis/funnel_analyzer.py

```
@@ -0,0 +1,473 @@
+"""Arc Funnel Analyzer - Capability Decomposition for Execution Flow.
+
+Bridges agent capabilities from ingestion to real-time execution analysis,
+providing step-by-step success rate tracking and bottleneck identification.
+"""
+
+import logging
+import asyncio
+import json
+from typing import Dict, List, Any, Optional, Tuple
```

Ruff (0.11.9)

`typing.Dict` is deprecated, use `dict` instead

(UP035)